application-admintools-ui/src/main/resources/AdminTools/Code/AdminToolsJS.xml

```diff
@@ -125,7 +125,94 @@
+    let documentReference = XWiki.Model.resolve('AdminTools.Code.HealthCheckResources', XWiki.EntityType.DOCUMENT);
+    return new XWiki.Document(documentReference).getURL('get');
+  };
+
+  const runHealthCheckJob = function() {
+    $('#admin_tools_job_start').prop("disabled", true);
+    var data = [];
+    data.push(
+      {name: 'outputSyntax', value: 'plain'},
+      {name: 'sheet', value: 'wiki:AdminTools.Code.HealthCheckResources'},
+      {name: 'action', value: 'healthCheck'},
+      {name: 'form_token', value: xwikiMeta.form_token}
+    );
+    return Promise.resolve(new JobRunner({
+      createStatusRequest: function(jobId) {
+        return {
+          url: getHealthCheckActionPageURL(),
+          data: {
+            outputSyntax: 'plain',
+            sheet: 'AdminTools.Code.HealthCheckResources',
+            data: 'jobStatus',
+            jobId: jobId.join('/')
+          }
+        };
+      }
+    }).run(getHealthCheckActionPageURL(), data)).catch((reason) =&gt; {
+      return Promise.reject(reason);
+    }).finally(() =&gt; {
+      $('#admin_tools_job_start').prop("disabled", false);
+    });
+  };
+
+  // Need to call job.min.js only after the job has started its execution and the result message has been changed for
+  // the progress bar. Otherwise, job.js won't see .ui-progress and will not initialize and run a new JobRunner
+  // with the requested configuration. To make sure the job .ui-progress is seen, we need to undefine and request the
+  // resource again.
+  const initialiseJobJS = function() {
+    $.get(getHealthCheckActionPageURL(), function(data) {
+      let jobJsURL = $(data).find('#jobJsURL').attr('data-jobJsURL');
+      require.config({
+        paths: {
+          'health-check-job': jobJsURL
+        }
+      });
+      require(['health-check-job'], function(){});
+    });
+  };
+
+  const recallJobJs = function() {
+    require.undef('health-check-job');
+    require(['health-check-job'], function(){});
+  }
+
+  const reinitialiseHealthCheckResources = function() {
+    $.get(getHealthCheckActionPageURL(), function(data) {
+      let progressBar = $(data).find('div.ui-progress');
+      $('#healthCheck').find('.successmessage, .warningmessage, .errormessage').replaceWith(progressBar);
+      recallJobJs()
+    });
+  };
+
+  $(document).ready(function() {
+    if ($('#healthCheck').length) {
+      initialiseJobJS();
+      let jobState = $('#healthCheckJobState').attr('data-jobState');
+      if (jobState == "RUNNING") {
+        runHealthCheckJob();
+        $('.job-status').show();
+      } else if (jobState == "FINISHED") {
+        $('.job-status').show();
+      }
+    }
+  });
+
+  $(document).on('click', '.log-item-warn, .log-item-error', function(event) {
+    $(this).children(".recommendation").toggle();
+  });
+
+  $(document).on('click', '#admin_tools_job_start', function(event) {
+    event.preventDefault();
+    $(".log").empty();
```

Comment:
Single quotes should be used. Also, this looks pretty generic. If this name comes from the job module and you have another job running on the current page, you need to be more specific (I suppose this is inside a container that has a healthCheck class / id)
Same for `$('.job-status')` . Check if there aren't others